solutions/cses-1682.mdx

```diff
@@ -108,4 +110,85 @@ int main() {
 ```
 
 </CPPSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class FlightRoutesCheck {
+	static int n, m;
+	static boolean[] vis;
+
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+		n = io.nextInt();
+		m = io.nextInt();
+
+		// Stores the graph we are given (or G)
+		ArrayList<ArrayList<Integer>> adj1 = new ArrayList<>();
+		// Stores the reverse of the graph we are given (or G')
+		ArrayList<ArrayList<Integer>> adj2 = new ArrayList<>();
+
+		for (int i = 0; i < n; i++) {
+			adj1.add(new ArrayList<>());
+			adj2.add(new ArrayList<>());
+		}
+
+		for (int i = 0; i < m; i++) {
+			int a = io.nextInt()-1;
+			int b = io.nextInt()-1;
+			adj1.get(a).add(b);
+			adj2.get(b).add(a);
+		}
+
+		// Run dfs to check if you can reach all other vertices from vertex 1
+		vis = new boolean[n];
+		dfs(0, adj1);
+		for (int i = 0; i < n; i++) {
+			// If some vertex wasn't visited, that means that we cannot reach
+			// all other vertices, so we return NO
+			if (!vis[i]) {
+				io.println("NO");
+				io.println(1 + " " + (i+1));
```

Comment:
```suggestion
				io.println(1 + " " + (i + 1));
```

Reply:
i think you missed this? https://github.com/cpinitiative/usaco-guide/pull/1616/files#r641886491

Reply:
Whoops yeah I think I missed that one

Reply:
Should be committed now